avocado-framework/avocado
# [RFC] Virt test compat layer POC

Implement a plugin that functions as a compatibility layer for virt-test inside avocado. So far it's implemented as a separate command `vt-run`, but that was requested to be changed. It uses avocado output classes and logging facilities, but it defines its own `Job` class, adapted to work with virt-test. Example output:

    $ scripts/avocado vt-run -t qemu --tests type_specific.io-github-autotest-qemu.migrate.default.tcp --keep-image --keep-guest-running --keep-image-between-tests
    SETUP: PASS (0.30 s)
    JOB ID     : ba8ab988cf1ac3ccadfef2aa3d14d500679bfbc1
    JOB LOG    : /home/lmr/avocado/job-results/job-2015-03-30T02.43-ba8ab98/job.log
    TESTS      : 1
    (1/1) type_specific.io-github-autotest-qemu.migrate.default.tcp: PASS (38.61 s)
    PASS       : 1
    FAIL       : 0
    TIME       : 38.69 s

The location of the virt-test libraries can be set on the config file. So far, what it can do:

* Use the avocado logs location
* Use avocado logging facilities

What it still can't do:

* Use avocado data dir instead of virt-test
* Use the HTML plugin, since the job wasn't ported to the `TestResult` avocado paradigm
* Use other output plugins (--xunit and --json) for the same reasons mentioned above.

Next steps:

* Work on the limitations above
* After that, see how the code transitions to the `run` command. The problems we see are related to how the `virt-test`/`avocado` tests are defined and executed, that is still quite different between the two projects.
